src/xbot/common/math/FieldPose.java

```diff
@@ -0,0 +1,127 @@
+package xbot.common.math;
+
+public class FieldPose {
+
+    private double m;
+    private double b;
+    private ContiguousHeading heading;
+    private final XYPair fieldPosition;
+    
+    public FieldPose(XYPair point, ContiguousHeading heading) {
+        m = degreesToSlope(heading.getValue());
+        this.fieldPosition = point;
+        this.heading = heading;
+        
+        // if we already have the y-intercept, avoid division by 0.
+        if ((point.x == 0) || (Math.abs(m) < 0.01)) {
+            b = point.y;
+        } else {
+            b = point.y / (point.x * m); 
+        }   
+    }
+    
+    public FieldPose(XYPair point, double slope, boolean positiveAngle) {
+        this.fieldPosition = point;
+        m = slope;
+        b = point.y - (point.x * m);
```

Comment:
Oh man, they are both supposed to be subtract... means my tests aren't very good if they missed this!